pkg/errox/utils.go

```diff
@@ -14,3 +20,41 @@ func IsAny(err error, targets ...error) bool {
 	}
 	return false
 }
+
+func x[T error](err error) T {
+	var e T
+	if errors.Is(err, e) {
+		return e
+	}
+	return e
+}
+
+// ConcealSensitive strips sensitive data from some known error types and
+// returns a new error if something has been concealed, and otherwise the
+// original error is returned.
+func ConcealSensitive(err error) error {
```

Comment:
I think it will be great if this supports wrapping so we have a single call `WrapSensitive("msg", err)` instead of `errors.Wrap(ConcealSensitive(err), "msg")`